### PR TITLE
Some improvements for packaged build [wip]

### PR DESCRIPTION
**Pull request by noiseless**
I would like to share you my suggestions for improving TG_OWT_PACKAGED_BUILD.

Here they are:

1) cherry-picked #66 && fixed merge conflict
2) This [patch](https://github.com/desktop-app/tg_owt/pull/82/commits/ddcb55f55cfabd223bcd9b157fd2f25b723574e1) is an adaptation of Vitaly Lipatov's work <lav@altlinux.ru> for the latest versions of tg_owt. These changes are a credit to him and are now, as far as I can see, already being [used](http://www.sisyphus.ru/ru/srpm/Sisyphus/libowt-tg/patches/1) to build alt linux packages.

The patch adds cmake rules for building tg_owt with libyuv from the system/packages.

3) In this [patch](https://github.com/desktop-app/tg_owt/pull/82/commits/b6e101f3bccf211f6fd987cd71a40702ee6c19ef) I fix the paths to the libyuv header files. [Similar patch](https://github.com/TelegramMessenger/tgcalls/pull/13) for tgcalls (separately they make little sense - packaged build will break).

Testing.

To test my changes I did the following:

1) I have done the build according to the "[Build instructions for Linux using Docker](https://github.com/telegramdesktop/tdesktop/blob/dev/docs/building-linux.md)".

To apply my changes, I replaced these [lines](https://github.com/telegramdesktop/tdesktop/blob/dev/Telegram/build/docker/centos_env/Dockerfile#L806) in `centos_env/Dockerfile` from
```
RUN git remote add origin $GIT/desktop-app/tg_owt.git
RUN git fetch --depth=1 origin b02478677baac6d563589f216800ff9cea0fd65c
```
to
```
RUN git remote add origin $GIT/noiseless/tg_owt.git
RUN git fetch --depth=1 origin c5fc0b3ac814b599c161fcc93e62e9a267c86662
```
and also edited [.gitmodules](https://github.com/telegramdesktop/tdesktop/blob/dev/.gitmodules#L85):
```
[submodule "Telegram/ThirdParty/tgcalls"]
    path = Telegram/ThirdParty/tgcalls
    url = https://github.com/noiseless/tgcalls.git
    branch = system-libyuv
```
The build was successful.
2) I did a packaged build under OpenBSD. Tested situations where libyuv is installed on the system and where it is not there, but everything you need is in `./src/third_party/libyuv`.

In the first case, tg_owt was built with libyuv from the package, in the second, from `./src/third_party/libyuv`, i.e. everything is as expected.

What do you think of my patches? If necessary, I am ready to improve them according to your wishes.
Perhaps I should test some other build variants?

Thanks anyway.

### CODE REVIEW

**Comment by ilya-fedin** [comment]
> * In the last [patch](https://github.com/noiseless/tg_owt/commit/c5fc0b3ac814b599c161fcc93e62e9a267c86662) I turn off `cmake/{libabsl,libopenh264,libusrsctp,libvpx,libyuv}.cmake` for packaged builds.

Oh. Each time someone does this, snap build stops to build. So this would need extensive testing across all the building configuration.

File: CMakeLists.txt [hunk 1]
```diff
@@ -77,14 +77,16 @@ include(cmake/init_target.cmake)
...
 include(cmake/external.cmake)
-include(cmake/libabsl.cmake)
-include(cmake/libopenh264.cmake)
 include(cmake/libpffft.cmake)
 include(cmake/librnnoise.cmake)
 include(cmake/libsrtp.cmake)
-include(cmake/libusrsctp.cmake)
-include(cmake/libvpx.cmake)
-include(cmake/libyuv.cmake)
+if (NOT TG_OWT_PACKAGED_BUILD)
```
Comment (ilya-fedin) [inline CMakeLists.txt:83]: This breaks the fallback logic when a library is not found

Comment (klemensn) [reply CMakeLists.txt:83]: Yes, that can't be right.
https://github.com/desktop-app/tg_owt/pull/85 should do the trick, though.

Comment (noiseless) [reply CMakeLists.txt:83]: Yes, this is wrong, as Ilya has already told me. 598697567eaa46aec7589b81689cc836ff99e4a4 should fix that.
(However, there are linking problems with snap pack builds there; I haven't had a chance to reproduce it yet, but I'm still going to)

**Comment by ilya-fedin** [comment]
Is there still any work? Maybe close the PR and re-open when you would have time?

**Comment by noiseless** [comment]
reasonably

**Review by Elonmusk678 — APPROVED** [review]

**Pull request closed without merging**